docs/en/ingest-management/security/logstash-certificates.asciidoc

```diff
@@ -1,7 +1,301 @@
+//connection here because IMO that's beyond the scope of this doc and may
+//confused cloud users who don't need to know this.
+
+[discrete]
+[[generate-logstash-certs]]
+== Generate custom certificates and private keys
+
+You can use whatever process you typically use to generate PEM-formatted
+certificates. The examples shown here use the `certutil` tool provided by {es}.
+
+//REVIEWERS: The following worked for me, and I saw people asking on discuss. Is
+//there any reason we should not recommend this to cloud users? 
+
+TIP: The `certutil` tool is not available on {ecloud}, but you can still use it
+to generate certificates for {agent} to {ls} connections. Just
+https://www.elastic.co/downloads/elasticsearch[download an {es} package],
+extract it to a local directory, and run the `elasticsearch-certutil` command.
+There's no need to start {es}!
+
+//REVIEWERS: I did not set up /etc/hosts. Please help me refine the steps here.
+//for example, do I really need to set --ip and --dns? Or is there a better way?
+//Because I'm running all on the same machine (and using docker for the agent
+//deployment, I might be missing something.)
+
+. Generate a certificate authority (CA). Skip this step if you want to use an
+existing CA.
++
+--
+[source,shell]
+----
+./bin/elasticsearch-certutil ca --pem
+----
+
+This command creates a zip file that contains the CA certificate and key you'll
+use to sign the certificates. Extract the zip file:
+
+image::images/ca-certs.png[Screen capture of a folder called ca that contains two files: ca.crt and ca.key]
+--
+
+. Generate a client SSL certificate signed by your CA. For example:
++
+--
+[source,shell]
+----
+./bin/elasticsearch-certutil cert \
+  --name client \
+  --ca-cert /path/to/ca/ca.crt \
+  --ca-key /path/to/ca/ca.key \
+  --dns your.host.name.here \
+  --ip 192.0.2.1 \
+  --pem
+----
+
+Where `dns` and `ip` specify the name and IP address of the client.
+
+Extract the zip file:
+
+image::images/client-certs.png[Screen capture of a folder called client that contains two files: client.crt and client.key]
+--
+
+. Generate a {ls} SSL certificate signed by your CA. For example:
++
+--
+[source,shell]
+----
+./bin/elasticsearch-certutil cert \
+  --name logstash \
+  --ca-cert /path/to/ca/ca.crt \
+  --ca-key /path/to/ca/ca.key \
+  --dns your.host.name.here \
+  --ip 192.0.2.1 \
+  --pem
+----
+
+Where `dns` and `ip` specify the name and IP address of the {ls} server.
+
+Extract the zip file:
+
+image::images/logstash-certs.png[Screen capture of a folder called logstash that contains two files: logstash.crt and logstash.key]
+--
+
+. Convert the {ls} key to pkcs8. For example, on Linux run:
++
+[source,shell]
+----
+openssl pkcs8 -inform PEM -in logstash.key -topk8 -nocrypt -outform PEM -out logstash.pkcs8.key
+----
+
+//REVIEWERS: The GitHub issue says that Linux must be used to convert
+//the key. Is this true? Is what I have here ^^ accurate and good enough for Win
+//users?
```

Comment:
I'm not sure about this. I was able to convert the key to `pkcs8` on my Mac without issue. I believe the `openssl` executable should still work as expected here on Windows based on a quick look around Google.